pkg/options/start.go

```diff
@@ -54,6 +63,33 @@ func resourceByType(resources pipelineResources, restype string) []string {
 	return []string{}
 }
 
+func (taskRunOpts *TaskRunOpts) UseTaskRunFrom(tr *v1beta1.TaskRun, cs *cli.Clients, tname string, taskKind string) error {
+	var (
+		trUsed *v1beta1.TaskRun
+		err    error
+	)
+	if taskRunOpts.Last {
+		trUsed, err = task.LastRun(cs, tname, taskRunOpts.CliParams.Namespace(), taskKind)
+		if err != nil {
+			return err
+		}
+	} else if taskRunOpts.UseTaskRun != "" {
+		trUsed, err = tractions.Get(cs, taskRunOpts.UseTaskRun, metav1.GetOptions{}, taskRunOpts.CliParams.Namespace())
+		if err != nil {
+			return err
+		}
+	}
+
+	if len(trUsed.ObjectMeta.GenerateName) > 0 {
+		tr.ObjectMeta.GenerateName = trUsed.ObjectMeta.GenerateName
+	} else {
+		tr.ObjectMeta.GenerateName = trUsed.ObjectMeta.Name + "-"
+	}
+	// Copy over spec from last or previous TaskRun to use same values for this TaskRun
+	tr.Spec = trUsed.Spec
+	return nil
+}
```

Comment:
how about extracting this function from `ClusterTask` and `Task` and placing it over here? This would be using the struct created [here](https://github.com/tektoncd/cli/pull/1156/files#diff-9218d1c50a757c61ef9c8dae70738d86R28-R33)

/cc @danielhelfand @piyush-garg

Reply:
I think it makes sense as far as where to place it. And agree that task start and clustertask start should use the same func.